scripts/openapi.py

```
@@ -0,0 +1,668 @@
+    json.JSONDecodeError
+        If the response is not valid JSON.
+    """
+    url = base_url.rstrip("/") + SCHEMA_ENDPOINT
+    if format == "yaml":
+        url += "?format=yaml"
+
+    headers: dict[str, str] = {"Accept": "application/json"}
+    if token:
+        headers["Authorization"] = f"Token {token}"
+
+    req = urllib.request.Request(url, headers=headers)
```

Ruff (0.15.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)